s3/src/main/scala/monix/connect/s3/S3.scala

```diff
@@ -52,60 +58,393 @@ import software.amazon.awssdk.services.s3.model.{
-  *
-  * It is built on top of the [[software.amazon.awssdk.services.s3]],
-  * which is the reason why all the methods expects an implicit instance
-  * of a [[S3AsyncClient]] to be in the scope of the call.
+  * Singleton object provides builders for [[S3]].
   *
   * ==Example==
   *
   * {{{
-  *   import java.time.Duration
-  *
-  *   import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider
-  *   import software.amazon.awssdk.regions.Region.AWS_GLOBAL
-  *   import software.amazon.awssdk.http.nio.netty.NettyNioAsyncHttpClient
-  *   import software.amazon.awssdk.services.s3.S3AsyncClient
-  *
-  *   // the exceptions related with concurrency or timeouts from any of the requests
-  *   // might be solved by rainsing the `maxConcurrency`, `maxPendingConnectionAcquire` or
-  *   // `connectionAcquisitionTimeout` from the underlying netty http client.
-  *   // see below an example on how to increase such values.
+  * val bucket = "my-bucket"
+  * val key = "my-key"
+  * val content = "my-content"
   *
-  *   val httpClient = NettyNioAsyncHttpClient.builder()
-  *     .maxConcurrency(500)
-  *     .maxPendingConnectionAcquires(50000)
-  *     .connectionAcquisitionTimeout(Duration.ofSeconds(60))
-  *     .readTimeout(Duration.ofSeconds(60))
-  *     .build()
+  * def runS3App(s3: S3): Task[Array[Byte]] = {
+  *   for {
+  *     _ <- s3.createBucket(bucket)
+  *     _ <- s3.upload(bucket, key, content.getBytes)
+  *     existsObject <- s3.existsObject(bucket, key)
+  *     download <- {
+  *       if(existsObject) s3.download(bucket, key)
+  *       else Task.raiseError(NoSuchKeyException.builder().build())
+  *     }
+  *   } yield download
+  * }
   *
-  *   // necessary for all the exposed methods in the [[S3]] object
-  *   implicit val s3AsyncClient: S3AsyncClient = S3AsyncClient
-  *     .builder()
-  *     .httpClient(httpClient)
-  *     .credentialsProvider(DefaultCredentialsProvider.create())
-  *     .region(AWS_GLOBAL)
-  *     .build()
+  * val f = S3.create().use(s3 => runS3App(s3)).runToFuture
   * }}}
   *
   */
-object S3 {
+object S3 { self =>
+
+  /**
+    * Creates a [[Resource]] that will use the values from a
+    * configuration file to allocate and release a [[S3]].
+    * Thus, the api expects an `application.conf` file to be present
+    * in the `resources` folder.
+    *
+    * @see how does the expected `.conf` file should look like
+    *      https://github.com/monix/monix-connect/blob/master/aws-auth/src/main/resources/reference.conf`
+    *
+    * @see the cats effect resource data type: https://typelevel.org/cats-effect/datatypes/resource.html
+    *
+    * @return a [[Resource]] of [[Task]] that allocates and releases [[S3]].
+    */
+  def create(): Resource[Task, S3] = {
+    Resource.make {
+      for {
+        clientConf  <- Task.fromTry(Try(AppConf.loadOrThrow))
+        asyncClient <- Task.now(AsyncClientConversions.fromMonixAwsConf(clientConf.monixAws))
```

Comment:
No need for `Task` here

Reply:
I have replaced with `Task.eval(AppConf.loadOrThrow)`, though I think `Task` was needed.

Reply:
I mean that the same as
```
asyncClient = AsyncClientConversions.fromMonixAwsConf(clientConf.monixAws)
```